pixeltable/iterators/document.py

```diff
@@ -29,44 +28,59 @@ class Separator(enum.Enum):
 
+    # pdf-specific metadata
+    page_number: int = 0
+    bbox: Optional[Dict[str,float]] = None
+
 
 @dataclasses.dataclass
 class DocumentSection:
     """A single document chunk, according to some of the splitting criteria"""
     text: Optional[str]
-    md: Optional[DocumentSectionMd]
+    metadata: Optional[DocumentSectionMetadata]
 
 
 class DocumentSplitter(ComponentIterator):
-    """"Iterator over pieces of a document"""
-    MD_COLUMN_TYPES = {
+    """Iterator over pieces of a document. The document is split into chunks based on the specified separators.
+    The iterator output tuples are of schema {'text': StringType()}, but can include additional metadata fields if specified
+    in the `metadata` argument as explained below.
+
+    Args:
+        `metadata`: which additional metadata fields to include in the output schema:
+             'title', 'headings' (HTML and Markdown), 'sourceline' (HTML), 'page_number' (PDF), 'bounding_box' (PDF).
+             The input can be a comma-separated string of these values eg. 'title,headings,sourceline'.
+        `separators`: which separators to use to split the document into rows. Options are:
+             'heading', 'paragraph', 'sentence', 'token_limit', 'char_limit'. As with metadata, this is can be a
+                comma-separated string eg. 'heading,token_limit'.
+        `limit`: the maximum number of tokens or characters in each chunk if 'token_limit' or 'char_limit' is specified.
+    """
+    METADATA_COLUMN_TYPES = {
         ChunkMetadata.TITLE: StringType(),
         ChunkMetadata.HEADINGS: JsonType(),
-        ChunkMetadata.SOURCELINE: IntType()
+        ChunkMetadata.SOURCELINE: IntType(),
+        ChunkMetadata.PAGE_NUMBER: IntType(),
+        ChunkMetadata.BOUNDING_BOX: JsonType(),
     }
 
     def __init__(
             self, document: str, *, separators: str, limit: int = 0, overlap: int = 0, metadata: str = '',
-            html_skip_tags: List[str] = None, tiktoken_encoding: Optional[str] = 'cl100k_base',
+            html_skip_tags: Optional[List[str]] = None, tiktoken_encoding: Optional[str] = 'cl100k_base',
             tiktoken_target_model: Optional[str] = None
     ):
         import bs4
         if html_skip_tags is None:
             html_skip_tags = ['nav']
-        with open(document, 'r', encoding='utf8') as fh:
-            s = fh.read()
-            self._doc_handle = get_document_handle(s)
-            assert self._doc_handle is not None
-        self._separators = [Separator[s.upper()] for s in separators.split(',')]
-        self._md_fields = [ChunkMetadata[m.upper()] for m in metadata.split(',')] if len(metadata) > 0 else []
+        self._doc_handle = get_document_handle(document)
+        assert self._doc_handle is not None
+        self._separators = [Separator[s.strip().upper()] for s in separators.split(',')]
+        self._metadata_fields = [ChunkMetadata[m.strip().upper()] for m in metadata.split(',')] if len(metadata) > 0 else []
```

Comment:
Why is the `if len(metadata) > 0 else []` necessary? Won't the behavior be identical if we simply remove that clause?